File: 0050-powx-n/0050-powx-n.js
const myPow = (x, n) => {
  if (n === 0) return 1;
  let i = Math.abs(n), res = 1;
  
  while (i > 0) {
    if (i % 2 === 0) {
      x *= x;
      i /= 2;
    } else {
      i--;
      res *= x;
    }
  }

  return n > 0 ? res : 1 / res;
};

File: 0050-powx-n/0050-powx-n.go
func myPow(x float64, n int) float64 {
  ans, i := 1.0, n
  if n < 0 { i = -i }
  
  for i > 0 {
    if i % 2 == 0 {
      x *= x
      i /= 2
    } else {
      i--
      ans *= x
    }
  }

  if n > 0 { return ans }
  return 1 / ans
}